### PR TITLE
[WIP] Content and navigation restructuring

### PR DESCRIPTION
**Pull request by aslakhellesoy**
I've been learning some new Hugo tricks, and had a look at the [Akka docs](https://doc.akka.io/docs/akka/current/index.html) - I really like the way it's organised.

I'm experimenting with a new, simpler way to organise content, similar to those docs. It takes advantage of some Hugo features that can generate navigation from the structure, without having to do it in the `config.toml` file. Instead we put metadata in each page's front matter. Proposed structure:

```
.
└── content
    ├── install
    │   ├── _index.md
    │   ├── cplusplus.md
    │   ├── java.md
    │   ├── javascript.md
    │   └── ruby.md
    ├── guides-and-tutorials
    ├── cucumber-api
    ├── gherkin-reference
    ├── bdd
    ├── tools
    └── community
```

Each directory represents a [section](https://gohugo.io/content-management/sections/), which is a grouping of related pages. Each section has an `_index.md` file with a tiny blurb, and a grid of big squares linking to each page within the section. A bit like [this](https://doc.akka.io/docs/akka/current/guide/index.html).

I'm also trying out a navigation menu where everything is in the left navigation, and not in the top nav. There are three levels of nesting. Whatever page you are on will be expanded, and the other navigation branches will be collapsed. For example, if you are currently on the Java Installation page, the navigation will look like this:

* Install
  * C++
  * Java
    * Maven
    * Gradle
  * JavaScript
* Guides & Tutorials
* Cucumber API
* Gherkin Reference
* BDD
* Tools
* Community

In other words:
* Directory
  * File
    * H1

I'm unsure if we should go into H2 and H3 as well - that might be too deep.

### CODE REVIEW

**Comment by mlvandijk** [comment]
Looks interesting. Will have to look into more details after the day job ;)

**Comment by mlvandijk** [comment]
Is this PR still relevant? It looks like these changes were also made on master?

**Comment by aslakhellesoy** [comment]
Yes, this is merged now
